src/ImageSharp/Memory/MemoryOwnerExtensions.cs

```diff
@@ -13,20 +14,37 @@ namespace SixLabors.ImageSharp.Memory
     /// </summary>
     internal static class MemoryOwnerExtensions
     {
+        /// <summary>
+        /// Gets a <see cref="Span{T}"/> from an <see cref="IMemoryOwner{T}"/> instance.
+        /// </summary>
+        /// <param name="buffer">The buffer</param>
+        /// <returns>The <see cref="Span{T}"/></returns>
+        [Pure]
```

Comment:
Any particular reason for adding the `Pure` attribute?

Reply:
I just really like to be as verbose and explicit as I can when writing code, and to decorate methods with as many attributes as possible. That includes eg. `[Pure]`, `[DoesNotReturn]`, `[NotNullWhen(bool)]`, etc. Having the code analyzer display all the appropriate hints and warnings when you do so make sthe whole developement experience much better and less error prone. For instance when using `[Pure]`, you make it clear that the method does not modify any state and that calling it without using the return is useless. This way you'll get a warning if you ever call it and you forget to eg. assign its return value to something. Saved me more than once before, highly recommend 😄

Of course, I can remove them here if you prefer to just do a separate PR to just pass over the entire repo and add the missing annotations. Or if you prefer not to use them at all for some other reason. Let me know!

Reply:
I think I'd like to do without tbh. They're not used in the runtime anywhere I can find and by my understanding it's possible to mark an impure method with the attribute which can lead to masking of potential issues.

Reply:
That is correct, the attribute is not enforced and you need to take the extra care to use it appropriately.
Remove in https://github.com/SixLabors/ImageSharp/pull/1314/commits/d8618052d17a5a50109e46541bb31a059d4de34a as requested 😊

Reply:
Thanks mate.